Assets/Scripts/Scheduler/Scheduler.cs

```diff
@@ -0,0 +1,241 @@
+﻿#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software,
+// and you are welcome to redistribute it under certain conditions; See
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using System.Xml.Schema;
+using System.Xml.Serialization;
+using UnityEngine;
+
+namespace Scheduler
+{
+    /// <summary>
+    /// Generic scheduler class for tracking and dispatching ScheduledEvents.
+    /// </summary>
+    public class Scheduler : IXmlSerializable
+    {
+        private static Scheduler instance;
+
+        public Scheduler()
+        {
+            this.Events = new List<ScheduledEvent>();
+            this.EventPrototypes = GenerateEventPrototypes();
+        }
+
+        public static Scheduler Current
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    instance = new Scheduler();
+                }
+
+                return instance;
+            }
+        }
+
+        public List<ScheduledEvent> Events { get; protected set; }
```

Comment:
If you return a copy, then you can't use the Events property internally in the Scheduler class when you modify the list.